share/p2p/shrexeds/client.go

```diff
@@ -42,52 +42,34 @@ func NewClient(host host.Host, opts ...Option) (*Client, error) {
 	}, nil
 }
 
-// RequestEDS requests the full ODS from one of the given peers and returns the EDS.
-//
-// The peers are requested in a round-robin manner with retries until one of them gives a valid
-// response, blocking until the context is canceled or a valid response is given.
+// RequestEDS requests the ODS from the given peers and returns the EDS upon success.
 func (c *Client) RequestEDS(
 	ctx context.Context,
 	dataHash share.DataHash,
-	peers peer.IDSlice,
+	peers ...peer.ID,
 ) (*rsmt2d.ExtendedDataSquare, error) {
 	req := &p2p_pb.EDSRequest{Hash: dataHash}
 
-	// requests are retried for every peer until a valid response is received
-	excludedPeers := make(map[peer.ID]struct{})
-	for {
-		// if no peers are left, return
-		if len(peers) == len(excludedPeers) {
-			return nil, errNoMorePeers
+	for _, to := range peers {
+		eds, err := c.doRequest(ctx, req, to)
+		if eds != nil {
```

Comment:
why not `err != nil`? Is it possible that non-nil error will be returned with valid eds?

Reply:
because the error cases are covered below

Reply:
Why not `err == nil`? I know it is usually either data or non-nil error returned, but if it is not a case, logic will be wrong.

Reply:
Discussed in sync, we return `nil, nil` sometimes so `if eds != nil`  is necessary